### DISCUSSION TITLE
kiali graph  traffic flow not working

### DISCUSSION THREAD

**Discussion by raykishan8** [kiali/kiali#7010] [Community meetings]
I setup kiali(1.73) with helm and istio version 1.18 with istioctl operator . Also implemented service mesh and service mesh flow is also working perfectly fine.

The problem i am facing is when i am seeing kiali graph its saying empty graph. But when i click on idle nodes then components are visible.

- **Kiali version:*1.73*
- **Istio version:*1.18*
- **Kubernetes impl:** 
- **Kubernetes version:** 
- **Other notable environmental factors:**

**Comment by josunect** [comment]
> I setup kiali(1.73) with helm and istio version 1.18 with istioctl operator . Also implemented service mesh and service mesh flow is also working perfectly fine.
> 
> The problem i am facing is when i am seeing kiali graph its saying empty graph. But when i click on idle nodes then components are visible.
> 
>     * **Kiali version:_1.73_
> 
>     * **Istio version:_1.18_
> 
>     * **Kubernetes impl:**
> 
>     * **Kubernetes version:**
> 
>     * **Other notable environmental factors:**

@raykishan8 There could be multiple reasons why the graph appears empty. You can find potential explanations in this [FAQ
](https://kiali.io/docs/faq/graph/#emptygraph). Please check this resource to see if it resolves your issue, and let us know.

**Comment by raykishan8** [comment]
i almost tried everything but graph only comes when i select idle nodes but in that case traffic flow can't be seen. 

Even service mesh is working prefectly fine all ingress and egress is working even prometheus metrics and grafana dashboard is visible for istio. 

even tried travel control dashboard , everything working except kiali flow graph

**Comment by josunect** [comment]
> i almost tried everything but graph only comes when i select idle nodes but in that case traffic flow can't be seen.
> 
> Even service mesh is working prefectly fine all ingress and egress is working even prometheus metrics and grafana dashboard is visible for istio.
> 
> even tried travel control dashboard , everything working except kiali flow graph

I would review if: 

- Are there any Prometheus metrics that Kiali uses to set up the graph? Here you can check how to verify it: https://kiali.io/docs/faq/graph/#queryprom
- Is there any error in the Kiali pod logs?

**Reply by raykishan8** [reply]
let me confirm with prometheus data and i will revert
thanks for looking on it.

**Reply by raykishan8** [reply]
i got the reason we have multiple prometheus , now the thing is can we have multiple prometheus as external service for kiali?
 if yes then how?

external_services:
  prometheus:
    url:

**Reply by josunect** [reply]
Kiali requires a single endpoint for metrics. A multi cluster environment, for example, will require to aggregate metrics, that can be achieved by configuring Prometheus federation.

**Reply by raykishan8** [reply]
ok got it we used thanos to do that and finally now everything working. Thanks for helping me out all these.

**Reply by josunect** [reply]
Thanks for the feedback, @raykishan8, I will close the discussion then.

**Comment by jmazzitelli** [comment]
_converting to discussion_

This kind of question is almost always answered with the tips provided in the FAQs that josune pointed to.

If you think it is a bug, you would need to provide more details. The Istio telemetry within Prometheus should be the first thing you should look at.

If you followed the instructions in these FAQs, tell us what you see. Also, looking at the Kiali pod logs and reporting any errors you see (if any) would be helpful (as josune mentioned).

* https://kiali.io/docs/faq/graph/#emptygraph
* https://kiali.io/docs/faq/graph/#queryprom
